app/views/boxes/_upload_csv_box.haml

```diff
@@ -0,0 +1,146 @@
+        tooltip.classList.remove("hidden")
+        setTimeout(() => tooltip.classList.add("hidden"), 5000)
+    }
+    else {
+        tooltip.classList.toggle("hidden")
+    }
+  }
+
+  // Add file button behaviour
+  function handleFileSelect(change_event) {
+    // We use the 1st file from the FileList list
+    let f = change_event.target.files[0];
+    let reader = new FileReader();
+
+    // Closure to capture the file information.
+    reader.onload = (function(f) {
+      return async(load_event) => {
+        // Parse the input file
+        var text =  load_event.target.result
+        var lines = text.split(/\r\n|\r|\n/); // tolerate both Windows and Unix linebreaks
+        
+        var fieldNames = ['Batch','Concentration','Distractor','Instructions']
+        var samples = []
+        var startParsing = false;
+        
+        var found_batches = new Set()
+        var not_found_batches = new Set()
+        
+        for (const line of lines) {
+          var fields = line.split(',')
+          if (startParsing && fields.length >= fieldNames.length) {
+            var sample = {}
+            fieldNames.forEach( async field => { sample[field] = fieldNames.indexOf(field) })
+            if (!found_batches.has(fields[0])) {
+              // Check if the batch number exists
+              var found = await fetch('/batches/existing_batch_number?context='+context+"&batch_number="+fields[0])
+                .then((response) => response.json())
+                .then((r) => { return r.message; })
+              
+              found.length > 0 ? found_batches.add(fields[0]) : not_found_batches.add(fields[0])
+            }
+            if (not_found_batches.has(fields[0])) continue;
+            samples.push(sample);
+          }
+          if ( !startParsing && _.isEqual(fields, fieldNames) ) startParsing = true;
+        }
+        
+        if (samples.length == 0) {
+          document.getElementById(`csv-file-error`).classList.remove(`hidden`);
+        } 
+        else {
+          // Create row from template with filename and upload info
+          var template = document.getElementById(`csvFileRow`).content;
+          var fragment = template.cloneNode(true);
+
+          fragment.querySelector(`.uploaded-samples-count`).textContent = `${samples.length} Samples ` ;
+          fragment.querySelector(`.upload-icon`).classList.add(`icon-checkmark`)
+          fragment.querySelector(`.file-name`).textContent = `${f.name}`;
+          
+          // Bind click event to remove button
+          var removeFileRow = function(evt) {
+              evt.currentTarget.upload_info.remove();
+              change_event.target.remove();
+              document.getElementById(`add-box-file`).style.display = 'block';
+          }
+          fragment.querySelector(".remove_file").upload_info = fragment.querySelector(`.upload_info`)
+          fragment.querySelector(".remove_file").addEventListener('click', removeFileRow, false)
```

Comment:
I feel like we could make the `removeFileRow` global, especially if we always render the file input, we don't need `change_event.target.remove()`. We could then call it directly from the HTML:

```haml
.remove_file{onclick: "removeFileRow"}
```

I'm not sure we need to assign the `upload_info` attribute above. Since there are only one row, it could just:
```js
document.querySelector("#uploaded-files .upload-info").remove()
```

Same for `showToolTip` below.